### PR TITLE
Complete the soversion granularity

### PR DESCRIPTION
**Pull request by lu-zero**

### CODE REVIEW

**Comment by lu-zero** [comment]
@kpcyrd could you please test and confirm that works for your purpose as well?

File: src/lib.rs [hunk 1]
```diff
@@ -5,3 +5,22 @@ pub mod config;
...
+    fn main_version(&self) -> String {
+        if self.major == 0 {
+            if self.minor == 0 {
+                format!("{}.{}.{}", self.major, self.minor, self.patch)
+            } else {
+                format!("{}.{}", self.major, self.minor)
+            }
+        } else {
+            format!("{}", self.major)
+        }
```
Comment (kpcyrd) [inline src/lib.rs:24]:
```suggestion
        match (self.major, self.minor, self.patch) {
            (0, 0, patch) => format!("0.0.{patch}"),
            (0, minor, _) => format!("0.{minor}"),
            (major, _, _) => format!("{major}"),
        }
```

File: src/install.rs [hunk 2]
```diff
@@ -146,12 +144,14 @@ impl UnixLibNames {
...
-            .arg(&self.with_full_ver)
-            .arg(install_path_lib.join(&self.with_major_ver));
-        let _ = ln_sf.status().unwrap();
+        if self.with_main_ver != self.with_full_ver {
+            let mut ln_sf = std::process::Command::new("ln");
+            ln_sf.arg("-sf");
+            ln_sf
+                .arg(&self.with_full_ver)
+                .arg(install_path_lib.join(&self.with_main_ver));
+            let _ = ln_sf.status().unwrap();
```
Comment (kpcyrd) [inline src/install.rs:153]: This panics if the `ln` binary couldn't be executed, and ignores the `ExitStatus` if `ln` returned an error (below for `self.canonical` too).

File: src/lib.rs (hunk 1 above)
Comment (lu-zero) [reply src/lib.rs:24]: It is a nice idea.

Comment (lu-zero) [reply src/lib.rs:24]: Did you test that it works for you though? :)

**Pull request merged**